File: app/src/main/java/ownmanager/in/fragmentruntimesample/viewModel/SharedViewModel.java
package ownmanager.in.fragmentruntimesample.viewModel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class SharedViewModel extends ViewModel {
    private MutableLiveData<CharSequence> text = new MutableLiveData<>();

    public LiveData<CharSequence> getText() {
        return text;
    }

    public void setText(CharSequence input) {
        text.setValue(input);
    }
}

File: app/src/main/java/ownmanager/in/fragmentruntimesample/viewModel/SharedViewModel.kt
package ownmanager.`in`.fragmentruntimesample.viewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class SharedViewModel : ViewModel() {
    private val text = MutableLiveData<CharSequence>()
    fun getText(): LiveData<CharSequence> {
        return text
    }

    fun setText(input: CharSequence) {
        text.value = input
    }
}